OsmAnd-java/src/net/osmand/util/MapUtils.java

```diff
@@ -274,7 +281,29 @@ public int compare(MapObject o1, MapObject o2) {
 			}
 		});
 	}
-	
+
+    /* Share/SEND location to another app for a specific action (message, QRCode, etc) */
+	public static void shareLocation(Context ctx, double latitude, double longitude, int zoom) {
```

Comment:
Ok, I'll remove Android classes from OsmAnd-java.

About the old menu, first some menu items didn't really work fully.  For example, the SMS menu item didn't work with my SMS app TextSecure, and it caused crashes on tablets with no SMS.  And not all relevant apps were supported using those menu items.

Also, this menu is not a standard design pattern for Android apps.  Share/`ACTION_SEND` menus are provided Android, and they are what users expect to see when clicking on something called "Share" along with the standard "Share" icon.

This new menu item allows the user to choose many different SMS, messaging, email, and QR Code options, all from the standard Share/`ACTION_SEND` menu.